basket/news/backends/braze.py

```diff
@@ -0,0 +1,181 @@
+import json
+import warnings
+from enum import Enum
+from urllib.parse import urljoin, urlparse, urlunparse
+
+from django.conf import settings
+
+import requests
+
+
+# Braze errors: https://www.braze.com/docs/api/errors/
+class BrazeBadRequestError(Exception):
+    pass  # 400 error (invalid request)
+
+
+class BrazeUnauthorizedError(Exception):
+    pass  # 401 error (invalid API key)
+
+
+class BrazeForbiddenError(Exception):
+    pass  # 403 error (invalid permissions)
+
+
+class BrazeNotFoundError(Exception):
+    pass  # 404 error (invalid endpoint)
+
+
+class BrazeRateLimitError(Exception):
+    pass  # 429 error (rate limit exceeded)
+
+
+class BrazeInternalServerError(Exception):
+    pass  # 500 error (Braze server error)
+
+
+class BrazeClientError(Exception):
+    pass  # any other error
+
+
+class BrazeEndpoint(Enum):
+    CAMPAIGNS_TRIGGER_SEND = "/campaigns/trigger/send"
+    USERS_EXPORT_IDS = "/users/export/ids"
+    USERS_TRACK = "/users/track"
+
+
+class BrazeClient:
+    def __init__(self, base_url, api_key):
+        urlbits = urlparse(base_url)
+        if not urlbits.scheme or not urlbits.netloc:
+            raise ValueError("Invalid base_url")
+        self.api_url = urlunparse((urlbits.scheme, urlbits.netloc, "", "", "", ""))
+
+        self.api_key = api_key
+        if not self.api_key:
+            warnings.warn("Braze API key is not configured", stacklevel=2)
+
+        self.active = bool(self.api_key)
+
+    def _request(self, endpoint, data=None):
+        """
+        Make a request to the Braze API.
+
+        @param endpoint: The Braze endpoint to call, from the BrazeEndpoint enum.
+        @param data: The data to send to the endpoint, as a dict.
+        @return: The response from the Braze API, as a dict.
+        @raises: BrazeBadRequestError: 400 error (invalid request)
+        @raises: BrazeUnauthorizedError: 401 error (invalid API key)
+        @raises: BrazeForbiddenError: 403 error (invalid permissions)
+        @raises: BrazeNotFoundError: 404 error (invalid endpoint)
+        @raises: BrazeRateLimitError: 429 error (rate limit exceeded)
+        @raises: BrazeInternalServerError: 500 error (Braze server error)
+        @raises: BrazeClientError: any other error
+
+        """
+        if not self.active:
+            return
```

Comment:
Question, non-blocking: do we want to log anything to flag this in logs?

Reply:
There's a one-time warning on start up if the API key isn't supplied that should suffice. I just wanted it to be callable but no-op so the code integration can exist without throwing an error.